.pre-commit-config.yaml

```diff
@@ -0,0 +1,45 @@
+ci:
+  autoupdate_schedule: "quarterly"
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.5.0
+  hooks:
+  - id: check-added-large-files
+  - id: check-case-conflict
+  - id: check-executables-have-shebangs
+  - id: check-symlinks
+  - id: check-toml
+  - id: check-yaml
+  - id: debug-statements
+  - id: destroyed-symlinks
+  - id: end-of-file-fixer
+    exclude: '\.(graphml)$'
+  - id: trailing-whitespace
+    exclude: '\.(pdb|gro|top|sdf|xml|cif|graphml)$'
+- repo: https://github.com/psf/black
+  rev: 24.2.0
+  hooks:
+  - id: black
+  - id: black-jupyter
+- repo: https://github.com/PyCQA/isort
+  rev: 5.13.2
+  hooks:
+  - id: isort
+- repo: https://github.com/econchick/interrogate
```

Comment:
The config is controlled in the pyproject.toml (so that way if you run the tool as a standalone thing, you use the same config as the one that runs on pre-commit)

```
[tool.interrogate]
fail-under = 0 
ignore-regex = ["^get$", "^mock_.*", ".*BaseClass.*"]
# possible values for verbose: 0 (minimal output), 1 (-v), 2 (-vv)
verbose = 2 
color = true
exclude = ["build", "docs"]
``` 
Config options are listed here: https://github.com/econchick/interrogate?tab=readme-ov-file#configuration

I don't want the tools to get in our way, so if we find its too annoying and gives us many false positives, we can drop it.